docs/reference/query-languages/esql/esql-lookup-join.md

```diff
@@ -0,0 +1,104 @@
+
+* Pull in environment or ownership details for each host to correlate your metrics data.
+* Quickly see if any source IPs match known malicious addresses.
+* Tag logs with the owning team or escalation info for faster triage and incident response.
+
+[`LOOKUP join`](/reference/query-languages/esql/esql-commands.md#esql-lookup-join) is similar to [`ENRICH`](/reference/query-languages/esql/esql-commands.md#esql-enrich) in the fact that they both help you join data together. You should use `LOOKUP JOIN` when:
+
+* Enrichment data changes frequently
+* You want to avoid index time processing
+* Working with regular indices
+* Need to preserve distinct matches
+* Need to match on any field in a lookup index
+
+## How the `LOOKUP JOIN` command works [esql-how-lookup-join-works]
+
+The `LOOKUP JOIN` command adds new columns to a table, with data from {{es}} indices.
+
+:::{image} ../../../images/esql-lookup-join.png
+:alt: esql lookup join
+:::
+
+$$$esql-source-index$$$
+
+Source index
+:   An index which stores enrich data that the `LOOKUP` command can add to input tables. You can create and manage these indices just like a regular {{es}} index. You can use multiple source indices in an enrich policy. You also can use the same source index in multiple enrich policies.
+
+## Example
+
+In the case where there are multiple matches on the index `LOOKUP JOIN` the output rows is the combination of each match from the left with each match on the right. 
+
+Image you have the two tables:
+
+**Left**
+
+|Key|Entry|
+| --- | --- |
+|1|A|
+|2|B|
+
+**Right**
+
+|Key|Language|TLD|
+|---|---|---|
+|1|English|CA|
+|1|English|null|
+|1|null|UK|
+|1|English|US|
+|2|German|AT,DE|
+|2|German|CH|
+|2|German|null|
+
+Running the following query would provide the results shown below.
+
+```esql
+FROM Left
+| WHERE Language IS NOT NULL // works and filter TLD UK
+| LOOKUP JOIN Right ON Key
+```
+
+|Key|Entry|Language|TLD|
+|---|---|---|---|
+|1|A|English|CA|
+|1|A|English|null|
+|1|A|null|UK|
+|1|A|English|US|
+|2|A|German|AT,DE|
+|2|B|German|CH|
+|2|B|German|null|
+
+::::{tip}
+`LOOKUP JOIN` does not guarantee the output to be in any particular order. If a certain order is required, users should use a [`SORT`](/reference/query-languages/esql/esql-commands.md#esql-sort) somewhere after the `LOOKUP JOIN`.
+
+::::
+
+## Prerequisites [esql-enrich-prereqs]
+
+To use `LOOKUP JOIN`, you must have:
+
+* Data types of join key and join field in the lookup index need to generally be the same - up to widening of data types, where e.g. `short,byte` are considered equal to `integer`. Also, text fields can be used on the left hand side if and only if there is an exact subfield whose name is suffixed with `.keyword`.
+
+## Limitations
+
+The following are the current limitations with `LOOKUP JOIN`
+
+* `LOOKUP JOIN` will be sucessfull if both left and right type of the join are both `KEYWORD` types or if the left type is of `TEXT` and the right type is `KEYWORD`.
```

Comment:
This is only true if the left hand side `TEXT` field has an exact `.keyword` subfield.

nit: the notions of "left" and "right" are not really defined for LOOKUP JOIN, I believe.

Reply:
On second read, this looks maybe a bit misplaced? This is mentioned in more detail in the prerequisites above.